### ISSUE TITLE
bwa-mem2 always returns status code 1

### ISSUE THREAD

**Issue by klmr** [bwa-mem2/bwa-mem2#11]
bwa-mem2 reproducibly returns the status code 1, which causes scripts and Makefiles to assume failure.

To reproduce:

```sh
wget http://ftp.ensembl.org/pub/current_fasta/saccharomyces_cerevisiae/dna/Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz
bwa-mem2 index Saccharomyces_cerevisiae.R64-1-1.dna.toplevel.fa.gz
echo $?
# 1
```

Same when running on a `gunzip`-ped FASTA file instead.

Or, using human (sorry):

```sh
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz
bwa-mem2 index hs37d5.fa.gz
echo $?
# 1
```

The same is true when invoking `bwa-mem2 mem`.

```sh
bwa-mem2 version
# 2.0pre1
uname -srvio
# Linux 4.15.0-55-generic #60-Ubuntu SMP Tue Jul 2 18:22:20 UTC 2019 x86_64 GNU/Linux
lsb_release -a
# No LSB modules are available.
# Distributor ID: Ubuntu
# Description:    Ubuntu 18.04.2 LTS
# Release:        18.04
# Codename:       bionic
```

**Comment by sanchit-misra** [comment]
It seems you are running the latest release code. This is a known bug in the latest release and was fixed. Unfortunately, we haven't yet had any release after that. Could you try running the latest code from the repository?

**Comment by klmr** [comment]
Cool, I’ll try the dev version ASAP. But note that I also observed this behaviour for mapping, not just indexing (though it seems harder to repro … I’ll see if I still observe it with the latest version and report back).

**Comment by klmr** [comment]
I’ve finally gotten around to trying the repo version (def6c75627df63b8723d79d1a7abdabac0627b53) and compiled it via `make multi` but unfortunately I cannot run it:

```
$ ~/tmp/bwa-mem2/bwa-mem2 mem data/index.fasta data/reads_R1.fastq.gz data/reads_R2.fastq.gz >out.sam
-----------------------------
Executing in AVX512 mode!!
-----------------------------
Ref file: ../original/bwa-mem2/index.fasta
Entering FMI_search
reference seq len = 179780839
Segmentation fault (core dumped)
```

The same command works with pre1 (except that command returns status code 1).

**Comment by yuk12** [comment]
Can you try with the latest commit. If it still does not work then please provide the input files. Thanks.

**Comment by klmr** [comment]
Great! That fixes it. However, I first needed to fix the Makefile (#28) to get the project to compile.
